hrc_speech_prediction/evaluate_tree_model.py

```diff
@@ -0,0 +1,373 @@
+        n = len(labels)
+        for t in test_X:
+            # curr_b = model
+            # curr_c = model
+            # curr_s = model
+            print("\nNEW TRIAL\n")
+            for u in t:
+                y = labels.pop(0)
+
+                both, s, _, b = combined_model.take_action(model="both",
+                                                           utter=u,
+                                                           return_probs=True)
+
+                speech = combined_model.pred_action(s)
+                context, c = context_model.take_action(model="context")
+
+                self.C_probs.append(c)
+                self.S_probs.append(s)
+                self.B_probs.append(b)
+
+                #print(y, speech, context, both)
+
+                score[0] += 1.0 * (speech in y)
+                score[1] += 1.0 * (context in y)
+                score[2] += 1.0 * (both in y)
+
+            combined_model.reset()
+            context_model.reset()
+        return score / n
+
+
+
+    def plot_random_probs(self, num_plots):
+        self.test_on_one_participant("speech", speech_eps=0.1)
+        rand_idx = random.sample([i for i in range(0, len(self.C_probs))], num_plots)
+
+        Cs = np.array(self.C_probs)
+        Bs = np.array(self.B_probs)
+        Ss = np.array(self.S_probs)
+
+
+        labels = self.comined_model._speech_model.actions
+        width = .3
+
+        #axes[0].legend(('Speech', 'Context', 'Both'))
+        for i in rand_idx:
+            fig, ax = plt.subplots(1, 1)
+            X = np.arange(len(Cs[0]))
+            c = Cs[i, :]
+            b = Bs[i, :]
+            s = Ss[i, :]
+
+            print(c)
+
+            y = self.Ys[i]
+            u = self.Us[i]
+
+            nrmlz = 1.0 / sum(b)
+
+            ax.bar(X-width, s, width=width, color='r', align='center')
+            ax.bar(X, c, width=width, color='b', align='center', tick_label=labels)
+            ax.bar(X+width, b * nrmlz, width=width, color='g', align='center')
+            rects = ax.patches
+            max_prob = max(b * nrmlz)
+
+            ax.legend(('Speech', 'Context', 'Both'))
+
+            #This draws a star above most probable action
+            for r in rects:
+                if r.get_height() == max_prob:
+                    ax.text(r.get_x() + r.get_width()/2,
+                            r.get_height() * 1.01, '*', ha='center', va='bottom', fontsize="12")
+
+            ax.text(self.comined_model._speech_model.actions.index(y), max_prob, "$", fontsize="12")
+            #plt.xticks(X, labels, rotation=70)
+            ax.set_title(u, fontsize="9")
+            plt.setp(ax.get_xticklabels(), rotation=30, horizontalalignment='right')
+            fig.tight_layout()
+            plt.savefig("/home/jake/catkin_ws/src/hrc_speech_prediction/figs/fig.{}.png".format(i))
+            plt.clf() 
+    
+
+
+
+
+    def _print_result_table(self, results, key_title):
```

Comment:
Obviously this would benefit from some refactoring. (Never copy paste before asking first if there is a way to re-factor…)